### DISCUSSION TITLE
zlib not available error with Python 2.7.18 and macos

### DISCUSSION THREAD

**Discussion by fgriberi** [community/community#26935] [Actions]
<p>I have a pipeline with <em>actions/setup-python</em> and <em>macos-latest</em> as runner. The pipeline works well, I got an artifact but when I run it, the following error is raised:</p>
<pre><code>zipimport.ZipImportError: can't decompress data; zlib not available
</code></pre>
<p>I have tried with setup-python versions v1/v2 and I got the same problem. Using a pyenv installation instead and installing <em>zlib</em> with <em>brew</em>, the artifact work as is expected.<br>
Any suggestion? Thx</p>

**Comment by Yanjingzhu** [comment]
<p>Would you mind sharing your workflow yml file and an example repo to reproduce this issue ? </p>

**Comment by fgriberi** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> sure, here is an example:</p>
<pre><code>run-package-macos:
   name: test
   runs-on: macOS-latest

   steps:
    - name: Checkout
      uses: actions/checkout@master
    - name: Set up Python 2.7
      uses: actions/setup-python@v1
      with:
        python-version: 2.7
    - name: install custom lib
      run: |
      	cd lib
    	pip install -r requirements.txt
        python setup.py install
        ...
    - name: build
      run: |
        ...
        python setup.py clean build
    ...
</code></pre>
<p>When I run the generated build, I got the zlib error. Same issue with python 3…8.  Using this following wa it works well:</p>
<pre><code>- name: install python with zlib wa
  run: |
        brew update
        brew install zlib
        export CFLAGS="-I$(brew --prefix openssl)/include -I$(xcrun --show-sdk-path)/usr/include"
        export LDFLAGS="-L$(brew --prefix openssl)/lib"
        brew install pyenv
        pyenv install 2.7
        pyenv global 2.7
</code></pre>
<p>Thxs a lot! :slight_smile:</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/fgriberi">@fgriberi</a></p>
<p>Setup python action will not install zlib . So you need to install zlib in another step and export some environment variables. The solution you found is great. It is the same as what’s tarrex said here:</p>
<p><a href="https://github.com/pyenv/pyenv/issues/1219#issuecomment-448658430" rel="nofollow noopener">https://github.com/pyenv/pyenv/issues/1219#issuecomment-448658430</a></p>
